x/fungible/keeper/msg_server_burn_fungible_module_asset.go

```diff
@@ -0,0 +1,93 @@
+package keeper
+
+import (
+	"context"
+
+	cosmoserrors "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+
+	"github.com/zeta-chain/node/cmd/zetacored/config"
+	"github.com/zeta-chain/node/pkg/constant"
+	authoritytypes "github.com/zeta-chain/node/x/authority/types"
+	"github.com/zeta-chain/node/x/fungible/types"
+)
+
+// BurnFungibleModuleAsset burns the zrc20 balance on the fungible module
+// If the zero address is provided, it burns the native ZETA held from the fungible module
+func (k msgServer) BurnFungibleModuleAsset(
+	goCtx context.Context,
+	msg *types.MsgBurnFungibleModuleAsset,
+) (*types.MsgBurnFungibleModuleAssetResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// check signer permission
+	err := k.GetAuthorityKeeper().CheckAuthorization(ctx, msg)
+	if err != nil {
+		return nil, cosmoserrors.Wrap(authoritytypes.ErrUnauthorized, err.Error())
+	}
+
+	// if the zero address is provided, burn the ZETA from the fungible module
+	if msg.Zrc20Address == constant.EVMZeroAddress {
+		// get the balance of the fungible module
+		balance := k.bankKeeper.SpendableCoin(ctx, types.ModuleAddress, config.BaseDenom)
+		if balance.IsZero() {
+			return nil, cosmoserrors.Wrapf(
+				types.ErrZeroBalance,
+				"no balance found for fungible module",
+			)
+		}
+		if err := k.bankKeeper.BurnCoins(
```

Comment:
Why not have an optional amount field? 
Otherwise, we might have to make sure there are not cctx's in progress so not to burn tokens for a deposit inadvertently

Reply:
>Otherwise, we might have to make sure there are not cctx's in progress so not to burn tokens for a deposit inadvertently

What do you mean by this? Minting and burning from the fungible module should be atomic, every amount that appears out of transactions should be considered surplus